firebase-perf/src/test/java/com/google/firebase/perf/metrics/validator/FirebasePerfTraceValidatorTest.java

```diff
@@ -189,6 +189,14 @@ public void testInvalidCustomAttribute() {
     trace.clearCustomAttributes().putCustomAttributes("ga_test", "value");
     assertThat(new FirebasePerfTraceValidator(trace.build()).isValidPerfMetric()).isFalse();
 
+    trace = trace.clone();
```

Comment:
Why do we clone instead of creawting a new valid trace metric? E.g. if we made some other modification to the trace that causes it to be invalid, no amount of changing the custom attributes would make it valid, so the tests could be passing incorrectly.

Reply:
Fair point. I have separate the tests and have got devoid of clone now.